packages/grpc-js/test/test-client.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Allow `any` data type for testing runtime type checking.
+// tslint:disable no-any
+import * as assert from 'assert';
+import * as path from 'path';
+
+import * as grpc from '../src';
+import { Server, ServerCredentials } from '../src';
+import { ServiceClient, ServiceClientConstructor } from '../src/make-client';
+import { sendUnaryData, ServerUnaryCall } from '../src/server-call';
+
+import { loadProtoFile } from './common';
+import { ConnectivityState } from '../src/channel';
+
+const clientInsecureCreds = grpc.credentials.createInsecure();
+const serverInsecureCreds = ServerCredentials.createInsecure();
+
+describe('Client', () => {
+  let server: Server;
+  let client: ServiceClient;
+
+  before(done => {
+    const protoFile = path.join(__dirname, 'fixtures', 'echo_service.proto');
+    const echoService = loadProtoFile(protoFile)
+      .EchoService as ServiceClientConstructor;
+
+    server = new Server();
+    server.addService(echoService.service, {
```

Comment:
Does this test rely on a service being added to the server? I don't think client.waitForReady depends on this, see https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md and https://grpc.github.io/grpc/node/grpc.Client.html#waitForReady

Reply:
No, `waitForReady` should not depend on the server having any services. In fact, the whole test file can be simplified significantly by omitting all of the protobuf stuff and directly using `grpc.Client`.

Reply:
I'll simplify the tests.

Reply:
I've made that change.